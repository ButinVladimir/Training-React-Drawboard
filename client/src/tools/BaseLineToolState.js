import BaseToolState from './BaseToolState';

class BaseLineToolState extends BaseToolState {
  constructor() {
    super();

    this.width = 1;
    this.color = '#000000';
  }
}

export default BaseLineToolState;
