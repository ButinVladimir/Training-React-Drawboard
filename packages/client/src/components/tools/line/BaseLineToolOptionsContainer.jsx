import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BaseLineToolOptions from './BaseLineToolOptions';
import BaseLineToolState from '../../../tools/BaseLineToolState';

const widths = [1, 2, 3, 4, 5, 10, 20, 50];

class BaseLineToolOptionsContainer extends Component {
  constructor(props) {
    super(props);

    this.onSelectWidth = this.onSelectWidth.bind(this);
    this.onSelectColor = this.onSelectColor.bind(this);

    const { toolState } = this.props;

    this.state = {
      width: toolState.width,
      color: toolState.color,
    };
  }

  onSelectWidth(width) {
    const { toolState } = this.props;
    const parsedWidth = parseInt(width, 10);

    this.setState({ width: parsedWidth });
    toolState.width = parsedWidth;
  }

  onSelectColor(color) {
    const { toolState } = this.props;

    this.setState({ color });
    toolState.color = color;
  }

  render() {
    const { width, color } = this.state;

    return (
      <BaseLineToolOptions
        width={width}
        color={color}
        widths={widths}
        onSelectWidth={this.onSelectWidth}
        onSelectColor={this.onSelectColor}
      />
    );
  }
}

BaseLineToolOptionsContainer.propTypes = {
  toolState: PropTypes.instanceOf(BaseLineToolState).isRequired,
};

export default BaseLineToolOptionsContainer;
