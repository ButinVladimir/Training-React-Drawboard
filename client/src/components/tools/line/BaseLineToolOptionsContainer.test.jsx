import React from 'react';
import { shallow, mount } from 'enzyme';
import BaseLineToolOptionsContainer from './BaseLineToolOptionsContainer';
import BaseLineToolState from '../../../tools/BaseLineToolState';

describe('BaseLineToolOptionsContainer', () => {
  it('renders properly', () => {
    const width = 3;
    const color = '#123';
    const toolState = new BaseLineToolState();
    Object.assign(toolState, {
      width,
      color,
    });

    const component = shallow(<BaseLineToolOptionsContainer
      toolState={toolState}
    />);
    const componentState = component.state();

    expect(component).toMatchSnapshot();
    expect(componentState).toEqual({
      width,
      color,
    });
  });

  it('handles onSelectWidth event', () => {
    const toolState = new BaseLineToolState();
    const width = 7;

    const component = mount(<BaseLineToolOptionsContainer
      toolState={toolState}
    />);
    component.find('select[name="width"]').simulate('change', { target: { value: width } });
    const componentState = component.state();

    expect(componentState).toMatchObject({ width });
    expect(toolState).toMatchObject({ width });
  });

  it('handles onSelectColor event', () => {
    const toolState = new BaseLineToolState();
    const color = '#abc';

    const component = mount(<BaseLineToolOptionsContainer
      toolState={toolState}
    />);
    component.find('input[name="color"]').simulate('change', { target: { value: color } });
    const componentState = component.state();

    expect(componentState).toMatchObject({ color });
    expect(toolState).toMatchObject({ color });
  });
});
