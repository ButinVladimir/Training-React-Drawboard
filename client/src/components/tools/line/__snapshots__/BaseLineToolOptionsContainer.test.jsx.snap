// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLineToolOptionsContainer renders properly 1`] = `
<BaseLineToolOptions
  color="#123"
  onSelectColor={[Function]}
  onSelectWidth={[Function]}
  width={3}
  widths={
    Array [
      1,
      2,
      3,
      4,
      5,
      10,
      20,
      50,
    ]
  }
/>
`;
